server/src/main/java/org/elasticsearch/index/mapper/MapperRegistry.java

```diff
@@ -70,14 +70,14 @@ public Map<String, RuntimeField.Parser> getRuntimeFieldParsers() {
      * Return a map of the meta mappers that have been registered. The
      * returned map uses the name of the field as a key.
      */
-    public Map<String, MetadataFieldMapper.TypeParser> getMetadataMapperParsers(Version indexCreatedVersion) {
-        if (indexCreatedVersion.onOrAfter(Version.V_8_0_0)) {
+    public Map<String, MetadataFieldMapper.TypeParser> getMetadataMapperParsers(IndexVersion indexCreatedVersion) {
+        if (indexCreatedVersion.onOrAfter(IndexVersion.V_8_0_0)) {
             return metadataMapperParsers;
-        } else if (indexCreatedVersion.major == 7) {
+        } else if (indexCreatedVersion.onOrAfter(IndexVersion.V_7_0_0)) {
```

Comment:
IndexVersion doesn't have major/minor numbers, but the order of checks here ensures it still does the same thing

Reply:
thanks, this looks correct to me.